nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -215,70 +215,38 @@ int ddxProcessArgument(int argc, char *argv[], int i)
     {
       if ((!strcmp(argv[j], "-display")) && (j + 1 < argc))
       {
-        envOptions = malloc(strlen(argv[j + 1]) + 1);
-
-        if (envOptions != NULL)
-        {
-          envOptions = strcpy(envOptions, argv[j + 1]);
-        }
-        #ifdef WARNING
-        else
+        if ((envOptions = strdup(argv[j+1])) == NULL);
         {
-          fprintf(stderr, "ddxProcessArgument: WARNING! failed string allocation.\n");
+          FatalError("malloc failed");
```

Comment:
Wouldn't it be good to mention what we tried to malloc here?

Reply:
Well, maybe. But if you happen to suffer from malloc failing you have other problems... ;-) Will extend these messages anyway.